scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -45,17 +45,30 @@
   [node-name->resource-map resource-keyword]
   (->> node-name->resource-map vals (map resource-keyword) (reduce +)))
 
+(defn total-gpu-resource
+  "Given a map from node-name->resource-keyword->amount,
+  returns a map from gpu model to count for all nodes."
+  [node-name->resource-map]
+  (->> node-name->resource-map vals (map :gpus) (apply merge-with +)))
+
 (defn generate-offers
   "Given a compute cluster and maps with node capacity and existing pods, return a map from pool to offers."
   [compute-cluster node-name->node node-name->pods]
   (let [compute-cluster-name (cc/compute-cluster-name compute-cluster)
         node-name->capacity (api/get-capacity node-name->node)
         node-name->consumed (api/get-consumption node-name->pods)
         node-name->available (pc/map-from-keys (fn [node-name]
-                                                 (merge-with -
+                                                 (api/merge-resource-map-collection -
                                                              (node-name->capacity node-name)
                                                              (node-name->consumed node-name)))
-                                               (keys node-name->capacity))]
+                                               (keys node-name->capacity))
+        ; Grab every unique GPU model being represented
+        gpu-models (set/union
+                     (->> node-name->capacity vals (map :gpus) (apply merge) keys set)
+                     (->> node-name->capacity vals (map :gpus) (apply merge) keys set))
```

Comment:
Should one of these lines be using `node-name->consumed`?

Reply:
yes good call!

Reply:
Why do we need this merge at all? You can't consume if there's no capacity. I.e., isn't capacity a superset of consumed?

Reply:
you're right scott, will edit that.